### PR TITLE
[ironic]: multiple dependency updates

### PR DESCRIPTION
**Pull request by sandzwerg**
* [mariadb][backup] Use resolve helper for backup_v2 secrets
* [mariadb] initdb secret is required during install
* [memcached] Use resolve for sasl-db secret string
* [mysql_metrics] Use resolve for mysql-metrics connection string
* [rabbitmq] remove unused unacknowledged alert from rename
* [utils] add helper to resolve secrets that need urlquery function
* utils: Resolve no-longer-used "identity.password_for_user"
* utils: Resolve username for mysql and F5 URls
* [utils] Add support for secrets-injector for db_url_pxc function
* [openstack/utils] Do not drop "host" from default_transport_url

### CODE REVIEW

**Comment by BerndKue** [comment]
Seems that the file openstack/ironic/ci/test-values.yaml needs some updating.

**Review by BerndKue — APPROVED** [review]

**Comment by sandzwerg** [comment]
hmm when I try to lint locally I get something else than the CI there I get

```

==> Linting openstack/ironic

[ERROR] templates/: template: ironic/templates/secrets.yaml:19:7: executing "ironic/templates/secrets.yaml" at <include (print .Template.BasePath "/etc/_inspector_secrets.conf.tpl") .>: error calling include: template: ironic/templates/etc/_inspector_secrets.conf.tpl:5:14: executing "ironic/templates/etc/_inspector_secrets.conf.tpl" at <include "resolve_secret" .Values.global.ironicServiceUser>: error calling include: template: ironic/charts/utils/templates/_hosts.tpl:20:3: executing "resolve_secret" at <include "_resolve_secret" (tuple . false)>: error calling include: template: ironic/charts/utils/templates/_hosts.tpl:12:35: executing "_resolve_secret" at <$str>: invalid value; expected string


engine.go:206: [INFO] Missing required value: .Values.global.registry is missing

engine.go:206: [INFO] Missing required value: .Values.global.dockerHubMirrorAlternateRegion is missing

engine.go:206: [INFO] Missing required value: .Values.global.dockerHubMirrorAlternateRegion is missing

engine.go:206: [INFO] Missing required value: .Values.global.dockerHubMirrorAlternateRegion is missing

engine.go:206: [INFO] Missing required value: User with key ironic requires password

Error: 1 chart(s) linted, 1 chart(s) failed
```

File: openstack/ironic/ci/test-values.yaml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 global:
+  registry: keppel.regionOne.cloud
+  dockerHubMirrorAlternateRegion: myRegistry/dockerhubalternate
   master_password: topSecret
   dbPassword: secret
   ipmi_exporter_user_passwd: topSecret
```
Comment (BerndKue) [inline openstack/ironic/ci/test-values.yaml:6]: Would it be possible to clean this up and either only use underscores or camelCase variables in the charts?

Comment (sandzwerg) [reply openstack/ironic/ci/test-values.yaml:6]: I think that depends on what the chart requires and not what the tests provide? So not sure I have the power to change this

**Pull request merged**